rpc/server.go

```diff
@@ -101,11 +102,17 @@ func NewServer(cfg *Config) (*Server, error) {
 		},
 	}
 
-	return &Server{
+	s := &Server{
 		ctx:        cfg.Ctx,
 		listener:   ln,
 		httpServer: server,
-	}, nil
+	}
+
+	if err = rpcServer.RegisterService(NewDaemonService(s, cfg), "daemon"); err != nil {
```

Comment:
In the interest of self documenting code, where it is easy to track who uses what information, we should probably pass down a separate config type with the minimal set of needed fields for `NewDaemonService`. If it was using the majority of the fields, it might not be worth the trouble, but I suspect that we are only using a small subset here.

Reply:
I ended up passing `cfg.ProtocolBackend` to the service as this is the only field that's used (same as the other services).